### PR TITLE
TF-2237 Limit composer send button clickable only one time

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
---
#2237 

## Resolved

https://github.com/linagora/tmail-flutter/assets/80142234/8aa9620f-3525-4541-ac7a-c0517113d1de

- Send with long email:

https://github.com/linagora/tmail-flutter/assets/80142234/59d999b0-b54e-430a-8a87-615063a5e7cd

### CODE REVIEW

**Comment by hoangdat** [comment]
Can you demo with the email which have very long content + inline image (which make the process of generating email long)? Thanks

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -288,6 +289,9 @@ class ComposerController extends BaseController {
     } else if (failure is GetEmailContentFailure ||
         failure is TransformHtmlEmailContentFailure) {
       emailContentsViewState.value = Left(failure);
+      if (isSendEmailLoading.value == true) {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:292]:
```suggestion
      if (isSendEmailLoading.isTrue) {
```

File: lib/features/composer/presentation/composer_controller.dart [hunk 2]
```diff
@@ -296,6 +300,9 @@ class ComposerController extends BaseController {
     super.handleExceptionAction(failure: failure, exception: exception);
     if (failure is GetAllIdentitiesFailure) {
       _handleGetAllIdentitiesFailure(failure);
+      if (isSendEmailLoading.value == true) {
+        isSendEmailLoading.value = false;
+      }
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:305]: IMO, We don't need loading sending button here, GetAllIdentities is only called when initializing composer.

File: lib/features/composer/presentation/composer_controller.dart [hunk 3]
```diff
@@ -787,8 +794,14 @@ class ComposerController extends BaseController {
   }
 
   void sendEmailAction(BuildContext context) async {
+    if (isSendEmailLoading.value == true) {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:797]:
```suggestion
    if (isSendEmailLoading.isTrue) {
```

File: lib/features/composer/presentation/composer_controller.dart [hunk 4]
```diff
@@ -800,7 +813,7 @@ class ComposerController extends BaseController {
       showConfirmDialogAction(context,
           AppLocalizations.of(context).message_dialog_send_email_without_recipient,
           AppLocalizations.of(context).add_recipients,
-          onConfirmAction: () => {},
+          onConfirmAction: () => {isSendEmailLoading.value = false},
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:816]: Please set `isSendEmailLoading = false` in `onCancelAction`

File: lib/features/composer/presentation/composer_controller.dart [hunk 5]
```diff
@@ -819,6 +832,7 @@ class ComposerController extends BaseController {
             toAddressExpandMode.value = ExpandMode.EXPAND;
             ccAddressExpandMode.value = ExpandMode.EXPAND;
             bccAddressExpandMode.value = ExpandMode.EXPAND;
+            isSendEmailLoading.value = false;
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:835]: Please set `isSendEmailLoading = false` in `onCancelAction`

File: lib/features/composer/presentation/composer_controller.dart [hunk 6]
```diff
@@ -842,7 +857,7 @@ class ComposerController extends BaseController {
           context,
           AppLocalizations.of(context).messageDialogSendEmailUploadingAttachment,
           AppLocalizations.of(context).got_it,
-          onConfirmAction: () => {},
+          onConfirmAction: () => {isSendEmailLoading.value = false},
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:860]: Please set `isSendEmailLoading = false` in `onCancelAction`

File: lib/features/composer/presentation/composer_controller.dart [hunk 7]
```diff
@@ -855,7 +870,7 @@ class ComposerController extends BaseController {
           AppLocalizations.of(context).message_dialog_send_email_exceeds_maximum_size(
               filesize(mailboxDashBoardController.maxSizeAttachmentsPerEmail?.value ?? 0, 0)),
           AppLocalizations.of(context).got_it,
-          onConfirmAction: () => {},
+          onConfirmAction: () => {isSendEmailLoading.value = false},
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:873]: Please set `isSendEmailLoading = false` in `onCancelAction`

File: lib/features/composer/presentation/composer_controller.dart [hunk 8]
```diff
@@ -1459,6 +1474,7 @@ class ComposerController extends BaseController {
   }
 
   void _closeComposerWeb({dynamic result}) {
+    isSendEmailLoading.value = false;
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:1477]: please also add it to close composer on mobile

File: lib/features/composer/presentation/composer_controller.dart (hunk 4 above)
Comment (hieutbui) [reply lib/features/composer/presentation/composer_controller.dart:816]: https://github.com/linagora/tmail-flutter/assets/80142234/31e41933-2e19-4b1b-b9fe-803ae53cb48b

There are some cases clicking on confirm button after `isSendEmailLoading` is true so i thought set `isSendEmailLoading = false` in `onConfirmAction` is better.

**Comment by dab246** [comment]
@hieutbui Please fix comment

File: core/lib/presentation/views/button/tmail_button_widget.dart [hunk 9]
```diff
@@ -37,6 +37,7 @@ class TMailButtonWidget extends StatelessWidget {
   final TextDirection iconAlignment;
   final int? maxLines;
   final MainAxisSize mainAxisSize;
+  final bool? isLoading;
```
Comment (dab246) [inline core/lib/presentation/views/button/tmail_button_widget.dart:40]: remove nullable type

**Review by dab246 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**